src/sql/workbench/services/notebook/browser/models/cell.ts

```diff
@@ -1023,4 +1047,33 @@ export class CellModel extends Disposable implements ICellModel {
 			}));
 		}
 	}
+
+	/**
+	 * Cache start state for any existing charts.
+	 * This ensures that data can be passed to the grid output component when a cell is re-executed
+	 */
+	private cacheChartStateIfExists(): void {
+		this.clearPreviousChartState();
+		// If a cell's source was changed, don't cache chart state
+		if (!this._preventNextChartCache) {
+			this._outputs?.forEach(o => {
+				if (dataResourceDataExists(o)) {
+					if (o.metadata?.azdata_chartOptions) {
+						this._previousChartState.push(o.metadata.azdata_chartOptions);
+					} else {
+						this._previousChartState.push(null);
```

Comment:
Is there a reason why we use null instead of undefined?

Reply:
`null` explicitly says that there is no value, whereas `undefined` may refer to a variable that just hasn't been given a value yet.

If this breaks convention, happy to change it; it just was natural for me to put `null` in here.

Reply:
We should generally not be using null in our own code. Nothing else in our code base uses that convention so having null is confusing and could potentially have adverse side effects (since we sometimes check specifically for undefined)